addon/webextension/background/main.js

```diff
@@ -30,10 +30,14 @@ window.main = (function () {
 
   chrome.browserAction.onClicked.addListener(function(tab) {
     if(tab.url.match(/about:(newtab|blank)/i)) {
-      sendEvent("goto-myshots", "about-newtab");
+      catcher.watchPromise(analytics.refreshTelemetryPref().then(() => {
```

Comment:
Pref lookups are cheap, it's just a hash table in effect. I don't know the cost of browser.runtime messaging though, @kmaglione would know that better than me.